core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -234,123 +206,23 @@ public void addInput(Page page)
-            filterMaxDistinctValues = Math.max(filterMaxDistinctValues, valueSet.size());
-        }
-        if (filterMaxDistinctValues > maxDistinctValues || filterSizeInBytes > maxFilterSizeInBytes) {
-            // The whole filter (summed over all columns) exceeds maxFilterSizeInBytes or a column contains too many distinct values
-            handleTooLargePredicate();
-        }
-    }
 
-    private void handleTooLargePredicate()
-    {
-        // The resulting predicate is too large
-        if (minMaxChannels.isEmpty()) {
-            // allow all probe-side values to be read.
-            dynamicPredicateConsumer.addPartition(TupleDomain.all());
-        }
-        else {
-            if (minMaxCollectionLimit < 0) {
-                handleMinMaxCollectionLimitExceeded();
-            }
-            else {
-                // convert to min/max per column for orderable types
-                for (int i = 0; i < minMaxChannels.size(); i++) {
-                    Integer channelIndex = minMaxChannels.get(i);
-                    BlockPositionComparison comparison = minMaxComparisons.get(i);
-                    Block block = blockBuilders[channelIndex].build();
-                    updateMinMaxValues(block, i, comparison);
-                }
-            }
+        if (minMaxCollectionLimit >= 0) {
+            minMaxCollectionLimit -= page.getPositionCount();
         }
-        // Drop references to collected values.
-        valueSets = null;
-        blockBuilders = null;
-    }
-
-    private void handleMinMaxCollectionLimitExceeded()
-    {
-        // allow all probe-side values to be read.
-        dynamicPredicateConsumer.addPartition(TupleDomain.all());
-        // Drop references to collected values.
-        minValues = null;
-        maxValues = null;
-    }
-
-    private void updateMinMaxValues(Block block, int channelIndex, BlockPositionComparison comparison)
-    {
-        checkState(minValues != null && maxValues != null);
-        int minValuePosition = -1;
-        int maxValuePosition = -1;
-
-        for (int position = 0; position < block.getPositionCount(); ++position) {
-            if (block.isNull(position)) {
-                continue;
-            }
-            if (minValuePosition == -1) {
-                // First non-null value
-                minValuePosition = position;
-                maxValuePosition = position;
-                continue;
-            }
-            if (comparison.compare(block, position, block, minValuePosition) < 0) {
-                minValuePosition = position;
-            }
-            else if (comparison.compare(block, position, block, maxValuePosition) > 0) {
-                maxValuePosition = position;
+        if (minMaxCollectionLimit < 0) {
+            for (int channelIndex = 0; channelIndex < channels.size(); channelIndex++) {
+                channelFilters[channelIndex].disableMinMax();
             }
         }
```

Comment:
This should move inside the `if (minMaxCollectionLimit >= 0)` block, it doesn't need to be called more than once

Reply:
I don't follow, please be more specific about what is "it"

Reply:
"it" is the whole `if (minMaxCollectionLimit < 0)` block. It looks like we will call it for every page after `minMaxCollectionLimit < 0` condition is met even though that's not needed.